<launch>
  <!-- Arguments -->
  <arg name="set_base_frame" default="base_footprint"/>
  <arg name="set_odom_frame" default="odom"/>
  <arg name="set_map_frame"  default="slam_map"/>
  <arg name="output_pose_topic" default="/slam/pose" />

  <!-- Gmapping -->
  <node pkg="gmapping" type="slam_gmapping" name="gmapping" output="screen">
    <param name="base_frame" value="$(arg set_base_frame)"/>
    <param name="odom_frame" value="$(arg set_odom_frame)"/>
    <param name="map_frame"  value="$(arg set_map_frame)"/>
    <rosparam command="load" file="$(find closedloop_nav_slam)/launch/slam/gmapping_params.yaml" />
    <remap from="map" to="/slam_map"/>
    <remap from="map_metadata" to="/slam_map_metadata"/>
  </node>

  <node pkg="closedloop_nav_slam" type="tf_to_pose_converter_node" name="tf_to_pose_converter" output="screen">
    <param name="transform_publish_period" value="0.05"/>
    <param name="source_frame" value="$(arg set_map_frame)"/>
    <param name="target_frame" value="$(arg set_base_frame)"/>
    <remap from="/output_pose_topic" to="$(arg output_pose_topic)"/>
  </node>
</launch>
